addon/content/messageFooter.jsx

```diff
@@ -2,39 +2,64 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-/* globals PropTypes, React, ActionButton */
+/* globals PropTypes, React, ActionButton, messageActions */
 /* exported MessageFooter */
 
 class MessageFooter extends React.PureComponent {
   constructor(props) {
     super(props);
-    this.action = this.action.bind(this);
+    this.onActionButtonClick = this.onActionButtonClick.bind(this);
   }
 
-  action(msg) {
-    msg.msgUri = this.props.msgUri;
-    msg.id = this.props.id;
-    this.props.dispatch(msg);
+  onActionButtonClick(msg) {
+    const payload = {
+      msgUri: this.props.msgUri,
+      shiftKey: msg.shiftKey,
+    };
+    let action = null;
+    switch (msg.type) {
+      case "draft":
+        action = messageActions.editDraft(payload);
+        break;
+      case "reply":
+        action = messageActions.msgReply(payload);
```

Comment:
Seeing as these are now all within `messageActions` could we just drop the `msg` from `msgReply` etc as that is already implied.

If we do that, we could also simplify this switch down to `action = messageActions[type](payload);` maybe?

Reply:
That was my original thought, but I thought that that might make the code too hard to follow. It's annoying that the payload that the actions take is not uniform...My original thought was that the `ActionButton` should be a connected component from the start, but I backed off because it wasn't clear what message data it would need to be passed.

Reply:
Ok, no problem, let's leave the switch as it is for now.

Could we still drop the `msg` part though?